examples/tv-app/tv-common/clusters/messages/MessagesManager.cpp

```diff
@@ -18,62 +18,95 @@
 #include "MessagesManager.h"
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <vector>
 
 using namespace std;
+using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters::Messages;
 using Message = chip::app::Clusters::Messages::Structs::MessageStruct::Type;
 
 // Commands
-void MessagesManager::HandlePresentMessagesRequest(
-    const chip::ByteSpan & messageId, const MessagePriorityEnum & priority,
-    const chip::BitMask<MessageControlBitmap> & messageControl, const chip::app::DataModel::Nullable<uint32_t> & startTime,
-    const chip::app::DataModel::Nullable<uint16_t> & duration, const chip::CharSpan & messageText,
-    const chip::Optional<chip::app::DataModel::DecodableList<MessageResponseOption>> & responses)
+void MessagesManager::HandlePresentMessagesRequest(const ByteSpan & messageId, const MessagePriorityEnum & priority,
+                                                   const BitMask<MessageControlBitmap> & messageControl,
+                                                   const DataModel::Nullable<uint32_t> & startTime,
+                                                   const DataModel::Nullable<uint16_t> & duration, const CharSpan & messageText,
+                                                   const Optional<DataModel::DecodableList<MessageResponseOption>> & responses)
 {
-    Message message{
-        // TODO: Enable id
-        chip::ByteSpan(), priority, messageControl, startTime, duration,
-        // TODO: Enable text
-        chip::CharSpan()
-        // TODO: Convert responses to Optional<chip::app::DataModel::List<const
-        // chip::app::Clusters::Messages::Structs::MessageResponseOptionStruct::Type>> message.responses = responses;
-    };
-
-    mMessages.push_back(message);
+    ChipLogProgress(Controller, "HandlePresentMessagesRequest message:%s",
+                    std::string(messageText.data(), messageText.size()).c_str());
+
+    CachedMessage * cachedMessage                     = new CachedMessage(messageId, priority, messageControl, startTime, duration,
+                                                                          std::string(messageText.data(), messageText.size()));
+    std::unique_ptr<CachedMessage *> cachedMessagePtr = std::make_unique<CachedMessage *>(cachedMessage);
+
+    if (responses.HasValue())
+    {
+        auto iter = responses.Value().begin();
+        while (iter.Next())
+        {
+            auto & response = iter.GetValue();
+
+            VerifyOrReturn(response.messageResponseID.HasValue() && response.label.HasValue(),
+                           ChipLogProgress(Controller, "HandlePresentMessagesRequest missing respond id or label"));
+
+            CachedMessageOption * option = new CachedMessageOption(
+                response.messageResponseID.Value(), std::string(response.label.Value().data(), response.label.Value().size()));
+
+            cachedMessage->AddOption(option);
+        }
```

Comment:
Again, this is missing a check for "decoding failed".

Reply:
added check